contracts/Comptroller/Diamond/facets/PolicyFacet.sol

```diff
@@ -207,6 +234,7 @@ contract PolicyFacet is IPolicyFacet, XVSRewardsHelper {
         uint256 actualRepayAmount, // solhint-disable-line no-unused-vars
         uint256 borrowerIndex // solhint-disable-line no-unused-vars
     ) external {
+        ensureUserAllowed(payer);
```

Comment:
I think this is not needed, because we are already checking the permission for repaying in the `repayBorrowAllowed` check (the "pre-hook")